subprojects/model-core/src/main/java/org/gradle/internal/instantiation/AbstractClassGenerator.java

```diff
@@ -119,6 +119,8 @@ public AbstractClassGenerator(Collection<? extends InjectAnnotationHandler> allK
             // Else, the generated class has been collected, so generate a new one
         }
 
+        List<CustomInjectAnnotationPropertyHandler> customAnnotationPropertyHandlers = new ArrayList<CustomInjectAnnotationPropertyHandler>(enabledAnnotations.size());
```

Comment:
IIUC, these are handlers for custom inject annotations (custom meaning != `@Inject`). So for me the name is correct.

Reply:
For me, the user in this case is the other project which wants to support annotations for service injection other than `@Inject`. So the annotations are user defined IMO.

I don't like `NamedInjectAnnotation`, given that the annotations don't really have a name. We could call them `GradleProvidedInjectAnnotation`s, so it may be possible at some point that plugin authors actually define their own annotations.

How about `ProvidedInjectAnnotation`? As opposed the the built-in `@Inject` annotation?

Reply:
`ProvidedInjectAnnotation` sounds good.